### PR TITLE
Categories (Explore Page)

### PR DESCRIPTION
**Pull request by cinquemb**
Will deprecate #122

### CODE REVIEW

File: app/models/category_favorite.rb [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  belongs_to :user, required: true
+  belongs_to :category, required: true
+
+  validates :user_id, uniqueness: {
+    scope: :category_id,
+    message: 'Cannot fave a category multiple times'
+  }
+
+  after_save do
+    client ||= Stream::CustomEndpointClient.new
```
Comment (NuckChorris) [inline app/models/category_favorite.rb]: `||=` doesn't make much sense here

File: app/services/trending_service.rb [hunk 2]
```diff
@@ -75,6 +77,24 @@ def trending_key(uid = nil)
...
+    key
+  end
+
+  def handle_trending_categories(id, weight = 1.0)
+    return unless %w[Anime Manga Drama].include?(namespace.name)
+    categories = namespace.includes(:categories).find_by(id: id).categories
+    categories.each do |category|
+      key = trending_category_key(category.id)
+      update_score(key, id, weight)
+      trim(key, limit: NETWORK_LIMIT) if rand < TRIM_PROBABILITY
```
Comment (NuckChorris) [inline app/services/trending_service.rb]: Should use a different constant from `NETWORK_LIMIT`, maybe `CATEGORY_LIMIT`?

File: lib/anidb/category_importer.rb [hunk 3]
```diff
@@ -0,0 +1,45 @@
+class AnidbCategoryImport
+  class ImportFile
```
Comment (NuckChorris) [inline lib/anidb/category_importer.rb]: I think we want to keep the class names matching the filename, so `anidb/category_importer.rb` would be `AniDB::CategoryImporter` and `AnidbCategoryImport::ImportFile` would be `anidb_category_import/import_file.rb` — though honestly you can just keep `ImportFile` inline, but probably want to rename the outermost class here to match the file.

File: lib/anidb/media_category_assoc_importer.rb [hunk 4]
```diff
@@ -0,0 +1,94 @@
+class AnidbAssocMediaCategoryImport
```
Comment (NuckChorris) [inline lib/anidb/media_category_assoc_importer.rb]: A namespace might be good? 🤔  Maybe `AnidbCategoryImport` (`lib/anidb_category_import/media_import.rb`, etc.)?

File: app/services/trending_service.rb (hunk 2 above)
Comment (cinquemb) [reply app/services/trending_service.rb]: Make sense to me, although i'm sure what would be appropriate to set it to. Should we keep it at 20?

File: lib/anidb/media_category_assoc_importer.rb (hunk 4 above)
Comment (cinquemb) [reply lib/anidb/media_category_assoc_importer.rb]: Sounds good to me

File: lib/anidb/category_importer.rb (hunk 3 above)
Comment (cinquemb) [reply lib/anidb/category_importer.rb]: Ok, i'll change the outer one.

**Review by trmcnvn — APPROVED** [review]
:+1:

**Pull request merged**